subsys/net/l2/ieee802154/ieee802154_mgmt.c

```diff
@@ -64,6 +65,10 @@ enum net_verdict ieee802154_handle_beacon(struct net_if *iface,
 				IEEE802154_EXT_ADDR_LENGTH);
 	}
 
+	header_length = ieee802514_beacon_header_length(mpdu->payload, mpdu->payload_length);
+	ctx->scan_ctx->payload_len = mpdu->payload_length - header_length;
```

Comment:
Now the "payload" prefix doesn't seem right any more. What about `beacon_payload_*`? Same reasoning as above. We might abbreviate to `beac_payload_*` but I personally find this ugly. See what you prefer.

Reply:
I like the full name better :)